components/server/src/authorization/spicedb.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { v1 } from "@authzed/authzed-node";
+
+export const SpiceDBClient = Symbol("SpiceDBClient");
+export type SpiceDBClient = v1.ZedPromiseClientInterface;
+
+export function spicedbClientFromEnv(): v1.ZedPromiseClientInterface {
+    const token = process.env["SPICEDB_PRESHARED_KEY"];
+    if (!token) {
+        throw new Error("No spicedb token configured.");
+    }
+
+    const address = process.env["SPICEDB_ADDRESS"];
+    if (!address) {
+        throw new Error("No spicedb address configured.");
+    }
+
+    const client = v1.NewClient(token, address, v1.ClientSecurity.INSECURE_PLAINTEXT_CREDENTIALS);
+    return client.promises;
+}
```

Comment:
Now that I think about it, do we need to guard for the case where the experimental config for SpiceDB is not set, and therefore SpiceDB doesn't exist? This could then prevent the instance from starting

Reply:
I've addressed this by making the type possibly undefined. The constructor no longer throws, when the env vars are undefined. The use of the client can then check.